pylint/checkers/similar.py

```diff
@@ -598,16 +598,16 @@ def stripped_lines(
     if ignore_imports or ignore_signatures:
         tree = astroid.parse("".join(lines))
     if ignore_imports:
-        node_is_import_by_lineno = (
-            (node.lineno, isinstance(node, (nodes.Import, nodes.ImportFrom)))
-            for node in tree.body
-        )
         line_begins_import = {
-            lineno: all(is_import for _, is_import in node_is_import_group)
-            for lineno, node_is_import_group in groupby(
-                node_is_import_by_lineno, key=lambda x: x[0]
-            )
+            node.lineno: True
+            for node in tree.nodes_of_class((nodes.Import, nodes.ImportFrom))
         }
+        line_begins_import.update(
+            {
+                node.lineno: isinstance(node, (nodes.Import, nodes.ImportFrom))
+                for node in tree.body
+            }
+        )
```

Comment:
If we update the fallback to the .get() call below to be False rather than `current_line_is_import` (sorry, my suggestion didn't really consider that), then we can do without this twice-iteration. That will probably obligate us to improve the variable name from `line_begins_import` 😄 .

Also, that should allow us to handle more deeply nested duplicate code like this, which so far I'm showing this PR is not handling:
```py
if typing.TYPE_CHECKING:
    if True:
        pass
    else:
        x = 1  # duplicate code begins... not visited by tree.body...
        y = 2
        z = 3
```

Reply:
Thank you for suggestion, implemented it. Code looks really clean now.

Unfortunately, unrelated test is failing due to API problem and should be restarted.